### PR TITLE
[JENKINS-52417] - Removing FilePath.copyFromRemotely 

### PR DESCRIPTION
**Pull request by jglick**
In favor of https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin/pull/9 as picked up by https://github.com/jenkinsci/workflow-basic-steps-plugin/pull/60 & https://github.com/jenkinsci/copyartifact-plugin/pull/104, the only places where this method had been called.

### Changelog entry

* RFE - Developer: Remove `hudson.FilePath#copyFromRemotely()` Beta API from the core

### CODE REVIEW

**Comment by daniel-beck** [comment]
Is this the first case of breaking updates due to Beta annotations?

**Comment by jglick** [comment]
> Is this the first case of breaking updates due to Beta annotations?

Yes, though the user impact should be zero since the calls to `copyFromRemotely` are only made when encountering a `VirtualFile` implementing `toExternalURL`, which only exists in `artifact-manager-s3`, which has never been released to the UC.

**Review by carlossg — APPROVED** [review]

**Review by kuisathaverat — APPROVED** [review]

**Review by daniel-beck — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Ready to merge once @jglick confirms it

**Comment by oleg-nenashev** [comment]
@jglick please follow the PR template and create a changelog entry proposal. Although the API is not used elsewhere, IMHO it is important to make it changelogged since the API is in LTS.

Maybe we also want to create a follow-up patch to 2.107.x Stable branch and change annotation to`@Restricted(DoNotUse.class)` there

**Comment by jglick** [comment]
All makes sense.

On hold until referenced PRs are merged.

**Comment by jglick** [comment]
jenkinsci/workflow-basic-steps-plugin#60 need not be merged first, since that is the same PR where this API was first used—has just been open that long. jenkinsci/copyartifact-plugin#104 ought to be merged and released first (implying a release of https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin/pull/9), since otherwise there could be an `artifact-manager-s3` release which offers external URLs which is used on a Jenkins without `copyFromRemotely` but `copyartifact` 1.40 which tries to call it, leading to a linkage error. @ikedam is maintainer, no response so far, so TBD.

**Comment by jglick** [comment]
https://github.com/jenkinsci/copyartifact-plugin/pull/104 has been merged but not yet released, so this is still on hold.

**Comment by jglick** [comment]
Trying to trigger a fresh build.

**Comment by oleg-nenashev** [comment]
Created https://issues.jenkins-ci.org/browse/JENKINS-52417 for it. IMHO tickets must be always required for Beta API removal. The change will need to be backported as discussed above

**Comment by oleg-nenashev** [comment]
Added the changelog entry

**Pull request merged**
